friends/src/App.js

```diff
@@ -0,0 +1,86 @@
+import React, { Component } from 'react';
+import './App.css';
+import axios from 'axios';
+
+class App extends Component {
+  constructor(props) {
+    super(props);
+  
+    this.state = {
+    friends: [],
+    name: '',
+    age: '',
+    email: '',
+  };
+}
+
+  componentDidMount() {
+    axios.get('http://localhost:5000/friends')
+      .then(response => {
+        this.setState({ friends: response.data});
+      })
+      .catch(err => {
+        console.log(err);
+      });
+  }
+
+  handleChange = event => {
+    this.setState({[event.target.name]: event.target.value})
+  }
+
+  saveFormData = () => {
+    const id = (this.state.friends[this.state.friends.length -1].id +1)
+    const nextFriend = { id: id, name: this.state.name, age: this.state.age, email: this.state.email };
+
+    axios.post(`http://localhost:5000/friends/`, nextFriend)
+    .then(response => {
+      this.setState({ friends: response.data, name: "", age: "", email: ""});
+    })
+    .catch(err => {
+      console.log(err);
+    });
+
+  } 
+
+  render() {
+    return (
+      <div className="App">
+      <h2>A list of friends, forthwith:</h2>
+        {this.state.friends.map(friend => {
+          return (
+            <div>
+            <h3>{friend.name}</h3>
+            <p>{friend.age}</p>
+            <p>{friend.email}</p>
+            </div>
+          )
+        })}
+         <div className="AddToFriendsForm">
```

Comment:
adding the form after the list causes the form to move every time a new friend is added...

think of the children (the users)! 😖